### PR TITLE
#313 Specify FT interceptor and the ordering policy

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang@uk.ibm.com>

### CODE REVIEW

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+[[fault-tolerance-interceptor]]
+
+== Fault Tolerance Interceptor
+
+The implementor of MicroProfile Fault Tolerance specification must provide a Fault Tolerance interceptor. The interceptor will be called if any one or many of the above interceptor bindings are specified. The default priority of the interceptor is `Priority.PLATFORM_AFTER+10`, which is `4010`. 
+
+The Fault Tolerance interceptor priority can be configured via MicroProfile Config with the property name of `mp.fault.tolerance.interceptor.priority`. The property value will only be read at application startup. Any subsequent value changes will not take effect until the application restarts.
+
+If a method is annotated with any of the Fault Tolerance interceptor bindings, it may specify other interceptor bindings. The bounded interceptors will be invoked in the ascending order of the interceptor priority, as specified by https://download.oracle.com/otn-pub/jcp/interceptors-1_2A-mrel3-eval-spec/Intercept.pdf?AuthParam=1541543428_889968d50f535735cbabe343350e09a4[Interceptor Specification^].
```
Comment (Azquelt) [inline spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]:
```suggestion
A method that is annotated with any of the Fault Tolerance interceptor bindings may also be annotated with other interceptor bindings. The bound interceptors will be invoked in ascending order of interceptor priority, as specified by https://download.oracle.com/otn-pub/jcp/interceptors-1_2A-mrel3-eval-spec/Intercept.pdf?AuthParam=1541543428_889968d50f535735cbabe343350e09a4[Interceptor Specification^]. 
```

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+== Fault Tolerance Interceptor
+
+The implementor of MicroProfile Fault Tolerance specification must provide a Fault Tolerance interceptor. The interceptor will be called if any one or many of the above interceptor bindings are specified. The default priority of the interceptor is `Priority.PLATFORM_AFTER+10`, which is `4010`. 
+
+The Fault Tolerance interceptor priority can be configured via MicroProfile Config with the property name of `mp.fault.tolerance.interceptor.priority`. The property value will only be read at application startup. Any subsequent value changes will not take effect until the application restarts.
+
+If a method is annotated with any of the Fault Tolerance interceptor bindings, it may specify other interceptor bindings. The bounded interceptors will be invoked in the ascending order of the interceptor priority, as specified by https://download.oracle.com/otn-pub/jcp/interceptors-1_2A-mrel3-eval-spec/Intercept.pdf?AuthParam=1541543428_889968d50f535735cbabe343350e09a4[Interceptor Specification^]. 
+
+For an instance, in the following example, MyLogInterceptor will be invoked first, followed by Fault Tolerance interceptor and then MyPrintInterceptor.
```
Comment (Azquelt) [inline spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]:
```suggestion
For instance, in the following example, `MyLogInterceptor` will be invoked first, followed by the Fault Tolerance interceptor and then `MyPrintInterceptor`.
```

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+[[fault-tolerance-interceptor]]
+
+== Fault Tolerance Interceptor
+
+The implementor of MicroProfile Fault Tolerance specification must provide a Fault Tolerance interceptor. The interceptor will be called if any one or many of the above interceptor bindings are specified. The default priority of the interceptor is `Priority.PLATFORM_AFTER+10`, which is `4010`.
```
Comment (Azquelt) [inline spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]:
```suggestion
The implementor of the MicroProfile Fault Tolerance specification must provide a Fault Tolerance interceptor. The interceptor will be called if one or more of the above interceptor bindings are specified. The default priority of this interceptor is `Priority.PLATFORM_AFTER+10`, which is `4010`. 
```

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc (hunk 1 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: fixed

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc (hunk 3 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: added one sentence.

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+[[fault-tolerance-interceptor]]
+
+== Fault Tolerance Interceptor
+
+The implementor of the MicroProfile Fault Tolerance specification must provide a Fault Tolerance interceptor. This interceptor provides the functionality for each Fault Tolerance annotation. The interceptor will be called if any one or many of the above interceptor bindings are specified. For instance, this interceptor will retry the specified operation if `Retry` annotation was specified on that operation. The default priority of the interceptor is `Priority.PLATFORM_AFTER+10`, which is `4010`.
```
[the remark this answers is not in the document]
Comment (carlosdlr) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: Another thing that we can change here is this part "The interceptor will be called if any one or many"  would be "The interceptor will be called if one or many of the above interceptor bindings are specified"

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc (hunk 2 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: fixed

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: rephrased.

**Comment by Emily-Jiang** [comment]
I have updated the doc again. Please recheck.

File: spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+The Fault Tolerance interceptor priority can be configured via MicroProfile Config with the property name of `mp.fault.tolerance.interceptor.priority`. The property value will only be read at application startup. Any subsequent value changes will not take effect until the application restarts.
+
+A method, annotated with any of the Fault Tolerance interceptor bindings, may also be annotated with other interceptor bindings. The bound interceptors will be invoked in ascending order of interceptor priority, as specified by https://download.oracle.com/otn-pub/jcp/interceptors-1_2A-mrel3-eval-spec/Intercept.pdf?AuthParam=1541543428_889968d50f535735cbabe343350e09a4[Interceptor Specification^]. If the appliction interceptors are enabled via beans.xml, the interceptors enabled via beans.xml will be invoked after the Fault Tolerance interceptor. For more details, refer to http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#enabled_interceptors[Interceptor ordering^] in CDI specification.
```
Comment (brunobat) [inline spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: The Oracle links try to track you and will not be accessible.
Please use: https://download.oracle.com/otn-pub/jcp/interceptors-1_2A-mrel3-eval-spec/Intercept.pdf
The JBoss one is ok.

Comment (Emily-Jiang) [reply spec/src/main/asciidoc/fault-tolerance-interceptor.asciidoc]: fixed. please double check.

**Pull request merged**